### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by omus**
Creates a new public repository for hosting a new GitHub action. This action will allow the currently running job to determine the job name and job ID allowing improved usage of the GitHub API.

A quick overview of the implementation: in order to determine the job ID we need to use the [GitHub API to list jobs for a workflow run attempt](https://docs.github.com/en/rest/actions/workflow-jobs?apiVersion=2022-11-28#list-jobs-for-a-workflow-run-attempt). In order to determine which job from the API results is the currently running job we need some kind of alternate way of uniquely selecting the job. I looked at the [GitHub Actions context information](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs) and the only overlap I found was the [`runner.name`](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/contexts#runner-context) context which according to the docs may not be unique. Alternatively, I chose to use the job name for the selection criteria which also isn't guaranteed to be unique but typically is and it is reasonable to require users to define unique job name within a user provided workflow. Unfortunately, the job name also isn't available via contexts so to get the job name we need to perform three steps:

1. Access the YAML workflow file which defined this currently running job
2. Use the `${{ github.job }}` which is the job YAML key in the workflow to determine the job name template (the job name as defined in YAML can contain GitHub expressions).
3. Render any GHA expressions in the job name template

With these steps we know the name of the job and can use that information to determine the job ID. If a job name happens to not be unique we can detect that and cause the action to fail and request that the user makes the job name unique.

In the future we may choose to use `runner.name` as a first pass in selecting the job from the GitHub API job list. I would consider this an optimization though as we would still need to the job name as either a fallback or as part of the selection criteria.

Various issues asking for the job name or ID feature:
- https://stackoverflow.com/questions/61611978/get-current-job-name-in-github-actions-workflow
- https://github.com/orgs/community/discussions/8945
- https://github.com/actions/runner/issues/324
- https://github.com/orgs/community/discussions/129314

### CODE REVIEW

**Comment by iamed2** [comment]
What's the motivation for this?

**Comment by omus** [comment]
> What's the motivation for this?

Knowing the `job_id` as used by the GitHub API is important for the new `matrix-output` GHA which requires this information to ensure that the last executed job in the matrix has a complete set of outputs. For more details see this design document: https://github.com/beacon-biosignals/matrix-output/pull/1/files#diff-be587c6fc5d417e1f9d10be50292738384b9df23b4699c8852c990612d3374a6

File: .github/workflows/integration-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+      contents: read
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./
+        id: job
+      - name: Test job name
+        run: '[[ "${output}" == "${expected}" ]] || exit 1'
+        env:
+          output: ${{ steps.job.outputs.name }}
+          expected: test-empty-name
```
Comment (kleinschmidt) [inline .github/workflows/integration-tests.yaml]: why this instead `${{ github.job }}` like the other null-y cases?

File: .github/workflows/integration-tests.yaml [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+      - name: Test job name
+        run: '[[ "${output}" == "${expected}" ]] || exit 1'
+        env:
+          output: ${{ steps.job.outputs.name }}
+          expected: Custom Name
+      - name: Test job ID
+        run: |
+          [[ "${job_id}" =~ ^[0-9]+$ ]] || exit 1
+        env:
+          job_id: ${{ steps.job.outputs.id }}
```
Comment (kleinschmidt) [inline .github/workflows/integration-tests.yaml:47]: can we test teh ID for the other cases as well?  it's not clear to me when it should return a real ID vs. some null/empty value

**Review by kleinschmidt — COMMENTED** [review]
this seems really really cursed but with reasonable test coverage I'm okay with it.

File: .github/workflows/integration-tests.yaml (hunk 2 above)
Comment (omus) [reply .github/workflows/integration-tests.yaml:47]: I'll update the tests.

File: .github/workflows/integration-tests.yaml (hunk 1 above)
Comment (omus) [reply .github/workflows/integration-tests.yaml]: Fair question. I think I'll make the other tests use a literal name and leave `${{ github.job }}` as a comment

**Review by kleinschmidt — APPROVED** [review]

**Comment by omus** [comment]
Thanks Dave. I'll merge this and make the first release on Monday!

**Pull request merged**